### PR TITLE
IDEMPIERE-5326: Parsing error for report view if where clause and vir…

### PR DESCRIPTION
**Pull request by dpansheriya**
…tual column used in print format

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/print/ReportEngine.java [hunk 1]
```diff
@@ -2345,11 +2412,6 @@ public String getWhereExtended() {
 		return m_whereExtended;
 	}
 
-	/* Save windowNo of the report to parse the context */
-	public void setWindowNo(int windowNo) {
-		m_windowNo = windowNo;
-	}
-
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/print/ReportEngine.java:2352]: should keep this for backward compatibility

**Review by hengsin — CHANGES_REQUESTED** [review]
please address the backward compatibility issue

File: org.adempiere.base/src/org/compiere/print/DataEngine.java [hunk 2]
```diff
@@ -86,18 +86,20 @@
...
 	
 	/**
 	 *	Constructor
 	 *	@param language Language of the data (for translation)
-	 *  @param trxName
+	 *	@param trxName
+	 *	@param windowNo
 	 */
-	public DataEngine (Language language, String trxName){
+	public DataEngine (Language language, String trxName, int windowNo){
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/print/DataEngine.java:98]: found another backward compatibility issue - please keep the existing public DataEngine (Language language, String trxName) constructor.

**Review by hengsin — CHANGES_REQUESTED** [review]

File: org.adempiere.base/src/org/compiere/print/layout/LayoutEngine.java [hunk 3]
```diff
@@ -112,21 +112,26 @@ public class LayoutEngine implements Pageable, Printable, Doc
...
 	 *	Detail Constructor
 	 *  @param format Print Format
 	 *  @param data Print Data
 	 *  @param query query for parameter info
+	 *  @param info
 	 *  @param trxName
+	 *  @param windowNo
 	 */
-	public LayoutEngine (MPrintFormat format, PrintData data, MQuery query, PrintInfo info ,  String trxName)
+	public LayoutEngine (MPrintFormat format, PrintData data, MQuery query, PrintInfo info ,  String trxName, int windowNo)
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/print/layout/LayoutEngine.java:132]: for backward compatibility, please keep the 2 existing LayoutEngine public constructor.

Comment (dpansheriya) [reply org.adempiere.base/src/org/compiere/print/layout/LayoutEngine.java:155]: This is resolved

File: org.adempiere.base/src/org/compiere/print/DataEngine.java (hunk 2 above)
Comment (dpansheriya) [reply org.adempiere.base/src/org/compiere/print/DataEngine.java:106]: This resolved

**Review by hengsin — APPROVED** [review]
looks good

**Pull request merged**